internal/overlord/stateengine.go

```diff
@@ -27,7 +27,7 @@ import (
 // the system and manipulating it to reflect the desired state.
 type StateManager interface {
 	// DryStart prepares the manager to run its activities without
-	// inccurring in side effects that might affect the system
+	// incurring in side effects that might affect the system
```

Comment:
Grammar nit (sorry, missed this previously).

```suggestion
	// incurring side effects that might affect the system
```